EasyLayout.Forms/EasyLayout.cs

```diff
@@ -567,112 +548,52 @@ private static int ConstantValue(Expression expr)
-            {
-                var memExpr = expr as MemberExpression;
-                if (memExpr == null)
-                {
-                    throw new NotSupportedException("Left hand side of a relation must be a member expression, instead it is " + expr);
-                }
-
-                position = GetPosition(memExpr);
-
-                viewExpr = memExpr.Expression as MemberExpression;
-                viewExpr = viewExpr?.Expression as MemberExpression;
-            }
-
-            if (viewExpr == null)
-            {
-                throw new NotSupportedException("Constraints should use views's Top, Bottom, etc properties, or extension methods like GetCenter().  Did you forget to include '.Bounds' in your clause?");
-            }
-
-            var view = GetView(viewExpr);
-
+            var viewExpr = fExpr?.Arguments.FirstOrDefault() as MemberExpression;
             return new LeftExpression
             {
-                View = view,
-                Position = position.Value,
-                Name = viewExpr.Member.Name
+                View = GetView(viewExpr),
+                Position = GetPosition(fExpr),
+                Name = viewExpr?.Member.Name
             };
         }
 
         private static object Eval(Expression expr)
         {
-            if (expr.NodeType == ExpressionType.Constant)
-            {
-                return ((ConstantExpression)expr).Value;
-            }
+            if (expr is ConstantExpression constExpr)
+                return constExpr.Value;
 
-            if (expr.NodeType == ExpressionType.MemberAccess)
-            {
-                var mexpr = (MemberExpression)expr;
-                var m = mexpr.Member;
+            if (expr is MemberExpression mexpr)
                 //todo: Where the heck is MemberType?
                 //if (m.MemberType == MemberTypes.Field)
                 //{
-                var fieldInfo = m as FieldInfo;
-                if (fieldInfo != null)
-                {
-                    var v = fieldInfo.GetValue(Eval(mexpr.Expression));
-                    return v;
-                }
-            }
+                if (mexpr.Member is FieldInfo fieldInfo)
+                    return fieldInfo.GetValue(Eval(mexpr.Expression));
 
-            if (expr.NodeType == ExpressionType.Convert)
+            if (expr.NodeType == ExpressionType.Convert && expr is UnaryExpression cexpr)
             {
-                var cexpr = (UnaryExpression)expr;
                 var op = Eval(cexpr.Operand);
-                if (cexpr.Method != null)
-                {
-                    return cexpr.Method.Invoke(null, new[] { op });
-                }
-                else
-                {
-                    return Convert.ChangeType(op, cexpr.Type);
-                }
+                return cexpr.Method != null
+                    ? cexpr.Method.Invoke(null, new[] {op})
+                    : Convert.ChangeType(op, cexpr.Type);
```

Comment:
I'm sorry, I'm just really not a big fan of ternaries that are this complicated.  I want to be able to easily set breakpoints when things go wrong.  Please consider reverting.